pymeasure/instruments/fakes.py

```diff
@@ -0,0 +1,200 @@
+            **kwargs
+        )
+        self._wait = wait
+        self._tstart = 0
+        self._voltage = 10
+        self._output_voltage = 0
+        self._time = 0
+        self._wave = self.wave
+        self._units = {'voltage': 'V',
+                       'output_voltage': 'V',
+                       'time': 's',
+                       'wave': 'a.u.'}
+        # mock image attributes
+        self._w = 1920
+        self._h = 1080
+        self._frame_format = "mono_8"
+
+    @property
+    def time(self):
+        """ Float property for elapsed time. """
+        if self._tstart == 0:
+            self._tstart = time.time()
+        self._time = time.time() - self._tstart
+        return self._time
+
+    @time.setter
+    def time(self, value):
+        if value == 0:
+            self._tstart = 0
+        else:
+            while self.time < value:
+                time.sleep(0.001)
+
+    @property
+    def wave(self):
+        """ Return a waveform.  """
+        return float(np.sin(self.time))
+
+    @property
+    def voltage(self):
+        """ Get the voltage. """
+        time.sleep(self._wait)
+        return self._voltage
+
+    @property
+    def output_voltage(self):
+        return self._output_voltage
+
+    @output_voltage.setter
+    def output_voltage(self, value):
+        """Set the voltage."""
+        time.sleep(self._wait)
+        self._output_voltage = value
+
+    @property
+    def frame_width(self):
+        """ Image frame width in pixels."""
+        time.sleep(self._wait)
+        return self._w
+
+    @frame_width.setter
+    def frame_width(self, w):
+        time.sleep(self._wait)
+        self._w = w
+
+    @property
+    def frame_height(self):
+        """ Image frame height in pixels."""
+        time.sleep(self._wait)
+        return self._h
+
+    @frame_height.setter
+    def frame_height(self, h):
+        time.sleep(self._wait)
+        self._h = h
+
+    @property
+    def frame_format(self):
+        """ Format for image data returned from the get_frame() method. Allowed values are:
+                mono_8: single channel 8-bit image.
+                mono_16: single channel 16-bit image.
+        """
+        time.sleep(self._wait)
+        return self._frame_format
+
+    @frame_format.setter
+    def frame_format(self, form):
+        allowed_formats = ["mono_8", "mono_16"]
+        strict_discrete_set(form, allowed_formats)
+        self._frame_format = form
+
+    @property
+    def frame(self):
+        """ Get a new image frame."""
+        im_format_maxval_dict = {"8": 255, "16": 65535}
+        im_format_type_dict = {"8": np.uint8, "16": np.uint16}
+        bit_depth = self.frame_format.split("_")[1]
+        time.sleep(self._wait)
+        return np.array(
+            im_format_maxval_dict[bit_depth] * np.random.rand(self.frame_height, self.frame_width),
+            dtype=im_format_type_dict[bit_depth]
+        )
+
+    def __getitem__(self, keys):
+        return keys
```

Comment:
Btw, I was wondering - what is this method for?

Reply:
I am not sure actually. It looks like that was implemented before I made any changes to the class. Shall we remove it?

Reply:
Should have checked myself instead of asking you, sorry. I'll remove that, I don't think it serves any purpose anymore.